packages/hardhat-router/internal/autosave-object.js

```diff
@@ -23,6 +24,8 @@ module.exports = function autosaveObject(file, initialState = {}) {
 
   const data = JSON.parse(fs.readFileSync(file));
 
+  const _write = debounce(() => write(file, data), 0);
```

Comment:
Debounce persitance to files to once every tick. This improves a lot the perfomance during saves in a single sync loop.